common4j/src/main/com/microsoft/identity/common/java/cache/MsalOAuth2TokenCache.java

```diff
@@ -1355,20 +1383,17 @@ private CredentialType getAccessTokenCredentialTypeForAuthenticationScheme(
     private boolean accountHasCredential(@NonNull final AccountRecord account,
                                          @NonNull final List<Credential> appCredentials) {
         final String methodName = ":accountHasCredential";
-
         final String accountHomeId = account.getHomeAccountId();
-        final String accountEnvironment = account.getEnvironment();
 
         Logger.verbosePII(
                 TAG + methodName,
                 "HomeAccountId: [" + accountHomeId + "]"
-                        + "\n"
-                        + "Environment: [" + accountEnvironment + "]"
         );
 
+        // Since we already filtered accounts and credentials by environment, there is no need to check
+        // environment again
```

Comment:
This assumes caller would have done this. And current callers do that, but as an independent util method, it should still match all environment.

Reply:
I can update the Javadoc to specify that this method does not filter by environment. Since the current callers already handle environment filtering, if there’s ever a need to filter by environment, a separate overloaded method can be created.